src/back/glsl/mod.rs

```diff
@@ -811,8 +811,22 @@ impl<'a, W: Write> Writer<'a, W> {
                 }
 
                 // Write the storage class
-                write!(self.out, "{} ", if output { "out" } else { "in" })?;
-
+                if self.options.version >= Version::Embedded(310)
+                    || self.options.version >= Version::Desktop(410)
+                {
+                    if let Binding::Location(index, ..) = *binding {
+                        write!(
+                            self.out,
+                            "layout(location = {}) {} ",
+                            index,
+                            if output { "out" } else { "in" }
+                        )?;
+                    } else {
+                        write!(self.out, "{} ", if output { "out" } else { "in" })?;
+                    }
+                } else {
+                    write!(self.out, "{} ", if output { "out" } else { "in" })?;
```

Comment:
Would you be interested to fix this properly for earlier GLSL versions? We basically need to have the naming scheme where the outputs from the previous stage are named the same as the inputs from the next stage.
I.e. these would be `_location_vs_fs` for VS outputs *and* FS inputs.

Reply:
Thank you for fix suggestions. Currently, I trying to fix this error for earlier GLSL versions. Just make a PR for a quick fix, because it is a blocker for `glsl-out`.

Reply:
no problem, I can follow up with this.